pkg/apiserver/user/shared/feature_flag.go

```diff
@@ -0,0 +1,15 @@
+// Copyright 2021 PingCAP, Inc. Licensed under Apache-2.0.
+
+package shared
+
+import "github.com/pingcap/tidb-dashboard/util/featureflag"
+
+type UserFeatureFlags struct {
+	NonRootLogin *featureflag.FeatureFlag
```

Comment:
I don't think this is something shared. It is SQL sign-in specific thing. I would rather see SSO to rely on something of SQL sign-in, instead of introducing SQL sign-in for all authenticators.